crates/core/component/ibc/src/component/msg_handler/connection_open_confirm.rs

```diff
@@ -69,23 +68,25 @@ impl MsgHandler for MsgConnectionOpenConfirm {
 
         // get the stored consensus state for the counterparty
         let trusted_consensus_state = state
-            .get_verified_consensus_state(self.proof_height_on_a, connection.client_id().clone())
+            .get_verified_consensus_state(self.proof_height_on_a, connection.client_id.clone())
             .await?;
 
         // PROOF VERIFICATION
         // in connectionOpenConfirm, only the inclusion of the connection state must be
         // verified, not the client or consensus states.
-        trusted_client_state.verify_connection_state(
+
+        let proof_conn_end_on_a = MerkleProof::try_from(self.proof_conn_end_on_a.clone())?;
+        proof_verification::verify_connection_state(
+            &trusted_client_state,
             self.proof_height_on_a,
-            connection.counterparty().prefix(),
-            &self.proof_conn_end_on_a,
-            trusted_consensus_state.root(),
-            &ConnectionPath::new(
-                connection
-                    .counterparty()
-                    .connection_id()
-                    .ok_or_else(|| anyhow::anyhow!("invalid counterparty"))?,
-            ),
+            &MerklePrefix {
+                key_prefix: connection.counterparty.prefix,
```

Comment:
perhaps counterparty.prefix should be a MerklePrefix type at the domain type level instead?